src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,103 @@
+import { Request, Response } from "express";
+
+import Cart from "../models/Cart";
+import {
+  mergeDuplicatedProduct,
+  getProductsWithQuantity
+} from "../utils/cartMergeProduct";
+
+interface CartItem {
+  productId: string;
+  Quantity: number;
+}
+export const createCart = async (req: Request, res: Response) => {
+  try {
+    const newuser: any = req.user;
+    const cartItems: CartItem[] = req.body;
+
+    const productsWithQuantity = await getProductsWithQuantity(cartItems);
+    let sum = 0;
+    productsWithQuantity.map((el) => {
+      sum += el.totalPrice;
+    });
+
+    const news = await mergeDuplicatedProduct(productsWithQuantity);
+    if (!news.StockCheck) {
+      return res.status(404).json({
+        error:
+          "you send the quantity which are more than what we have in the stock Please check."
```

Comment:
What does this message mean? What are you asking the user to check?

Reply:
The issue is now resolved.